### ISSUE TITLE
Placing a member element inside an array doesn't produce a warning

### ISSUE THREAD

**Issue by kylef** [apiaryio/drafter#313]
The following API Blueprint:

```apib
# Person [GET /person]
+ Response 200 (application/json)
    + Attributes
        + name: Kyle
        + addresses (array)
            + addresses (Address)


# Data Structures
## Address
+ city: London
```

Results in no warnings or errors and returns a data structure when parsed with Drafter 2.3.0 which looks like the following:

```yaml
element: "dataStructure"
content:
  -
	element: "object"
	content:
	  -
		element: "member"
		content:
		  key:
			element: "string"
			content: "name"
		  value:
			element: "string"
			content: "Kyle"
	  -
		element: "member"
		content:
		  key:
			element: "string"
			content: "addresses"
		  value:
			element: "array"
			content:
			  -
				element: "Address"
				content:
				  -
					element: "string"
					content: "addresses"
```

This data structure shows an array element containing an Address element, as created from `+ addresses (array)` then `+ addresses (Address)` from our blueprint.

Inside the Address element, we have content which is a string element with the value `addresses`. This element is a string and not a sample value which matches it's super type (Address/object) and thus invalid.

I think the example blueprint I have shown should result in a warning because the user has tried to place a member element inside an array and the subsequent refract data structure isn't valid.

The user probably meant either of the following:

```apib
 + addresses (array[Address])
```

```apib
+ addresses (array)
    + (Address)
```

and not:

```apib
+ addresses (array)
    + (Address)
```

**Comment by pksunkara** [comment]
@kylef I think this will be fixed in #246. /cc @w-vi

**Comment by pksunkara** [comment]
This is still a bug. Not fixed by #322 /cc @klokane

**Comment by kylef** [comment]
This has been resolved in current master.

```
$ ./bin/drafter drafter-313.apib -u -l

OK.
warning: (8)  array member definition of type 'object' contains value. You should use type definition without value eg. '- (object)'; line 6, column 13 - line 6, column 34
```
